### PR TITLE
Fix: crash with malformed ooni/run URL

### PR DESCRIPTION
**Pull request by aanorbel**
Fixes https://github.com/ooni/probe/issues/1978

## Proposed Changes

  - Update logic to validate `testArguments`

### CODE REVIEW

File: ooniprobe/View/OONIRun/OoniRunViewController.m [hunk 1]
```diff
@@ -166,11 +166,18 @@ - (void)showTestScreen{
...
-        if ([url length] < 2083 && [self validateLink:url]){
-            [Url checkExistingUrl:url];
-            [urls addObject:url];
+        @try {
+            if ([url length] < 2083 && [self validateLink:url]){
+                [Url checkExistingUrl:url];
+                [urls addObject:url];
+            }
+        } @catch (NSException *exception){
+            NSLog(@"OoniRunViewControllerError: %@", exception);
```
Comment (hellais) [inline ooniprobe/View/OONIRun/OoniRunViewController.m:180]: Should we perhaps submit this exception to sentry?

File: ooniprobe/View/OONIRun/OoniRunViewController.m [hunk 2]
```diff
@@ -178,6 +178,7 @@ -(void)validateAndAddURLs{
             }
         } @catch (NSException *exception){
             NSLog(@"OoniRunViewControllerError: %@", exception);
+            [ThirdPartyServices recordError:@"failure"  reason:exception.reason userInfo:exception.userInfo];
```
Comment (hellais) [inline ooniprobe/View/OONIRun/OoniRunViewController.m:181]: Will the reported exception include the OONI Run link that lead to the failure?

What is `userInfo`? Does this contain any personal information about the user submitting the exception?

Comment (aanorbel) [reply ooniprobe/View/OONIRun/OoniRunViewController.m:181]: `userInfo` is a dictionary containing application-specific data pertaining to the receiver.

For more info, see https://developer.apple.com/documentation/foundation/nsexception

Comment (hellais) [reply ooniprobe/View/OONIRun/OoniRunViewController.m:181]: Can we include also the OONI run link in the exception? I think it's going to be useful to be able to debug issues with it.

**Review by bassosimone — APPROVED** [review]
🐳

**Comment by bassosimone** [comment]
This PR is approved; @aanorbel can we merge this?

**Pull request merged**